tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js

```diff
@@ -0,0 +1,183 @@
+
+import styles from 'irene/components/ak-select/index.scss';
+import ENUMS from 'irene/enums';
+
+const classes = {
+  dropdown: styles['ak-select-dropdown'],
+  trigger: styles['ak-select-trigger'],
+  triggerError: styles['ak-select-trigger-error'],
+};
+
+module(
+  'Integration | Component | project-settings/general-settings/device-preferences-automated-dast',
+  function (hooks) {
+    setupRenderingTest(hooks);
+    setupMirage(hooks);
+    setupIntl(hooks);
+
+    hooks.beforeEach(async function () {
+      this.server.createList('organization', 1);
+
+      const store = this.owner.lookup('service:store');
+
+      const profile = this.server.create('profile', {
+        id: '1',
+      });
+
+      const file = this.server.create('file', {
+        project: '1',
+        profile: profile.id,
+      });
+
+      const project = this.server.create('project', {
+        file: file.id,
+        id: '1',
+        platform: ENUMS.PLATFORM.ANDROID,
+      });
+
+      this.server.get('/profiles/:id', (schema, req) =>
+        schema.profiles.find(`${req.params.id}`)?.toJSON()
+      );
+
+      this.server.get('/profiles/:id/device_preference', (schema, req) => {
+        return schema.devicePreferences.find(`${req.params.id}`)?.toJSON();
+      });
+
+      this.server.put('/profiles/:id/ds_automated_device_preference', () => {
+        return {
+          ds_automated_device_selection:
+            ENUMS.DS_MANUAL_DEVICE_SELECTION.ANY_DEVICE,
+        };
+      });
+
+      this.server.get(
+        '/profiles/:id/ds_automated_device_preference',
+        (schema) => {
+          console.log(schema);
+          return {
+            ds_automated_device_selection:
+              ENUMS.DS_MANUAL_DEVICE_SELECTION.ANY_DEVICE,
+            ds_automated_platform_version_min: 13,
+          };
+        }
+      );
+
+      this.server.get('/projects/:id/available-devices', (schema) => {
+        const results = schema.projectAvailableDevices.all().models;
+
+        return { count: results.length, next: null, previous: null, results };
+      });
+
+      const normalizedProject = store.normalize('project', {
+        ...project.toJSON(),
+      });
+
+      this.setProperties({
+        project: store.push(normalizedProject),
+      });
+
+      await this.owner.lookup('service:organization').load();
+    });
+
+    test('it renders', async function (assert) {
+      await render(hbs`
+      <ProjectSettings::GeneralSettings::DevicePreferencesAutomatedDast @project={{this.project}}/>
+    `);
+
+      assert
+        .dom(
+          '[data-test-projectSettings-generalSettings-devicePreferenceAutomatedDast-title]'
+        )
+        .exists()
+        .containsText('t:devicePreferencesAutomatedDast:()');
+
+      assert
+        .dom(
+          '[data-test-projectSettings-generalSettings-devicePreferenceAutomatedDast-preference-select]'
+        )
+        .exists();
+    });
+
+    test('it selects criteria', async function (assert) {
```

Comment:
Test does not check if selected device criteria and selection is being sent correctly via the network.